### PR TITLE
LPD-23775 - Create playwright tests to check if script cannot be added in a Workflow definition when script management configuration is disabled

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://liferay.atlassian.net/browse/LPD-23775

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:sf

**Comment by cgmonte** [comment]
@danvb112 this first commit is the one I have already reviewed in the other PR, right? Have you changed anything?

**Comment by danvb112** [comment]
Yes @cgmonte, this first commit has already been reviewed in another PR, but I need it to do some tests involving Notifications in Workflow, that's why it's here too, there's no need to review it.

File: modules/test/playwright/tests/portal-workflow-kaleo-designer-web/assignments.spec.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	await processBuilderPage.clickWorkflowDefinitionName(
+		workflowDefinitionName
+	);
+
+	await diagramViewPage.clickUpdateNodeLink();
+
+	await nodePropertiesSidebarPage.editAssignmentButton.click();
+
+	expect(
+		await page.$(`#assignment-type option[value="scriptedAssignment"]`)
```
Comment (cgmonte) [inline modules/test/playwright/tests/portal-workflow-kaleo-designer-web/assignments.spec.ts:60]: The use of element handles is [discouraged by Playwright](https://playwright.dev/docs/api/class-elementhandle). Have you tried using Locators?

File: modules/test/playwright/tests/portal-workflow-kaleo-designer-web/assignments.spec.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
```
Comment (cgmonte) [inline modules/test/playwright/tests/portal-workflow-kaleo-designer-web/assignments.spec.ts:86]: The use of element handles is [discouraged by Playwright](https://playwright.dev/docs/api/class-elementhandle). Have you tried using Locators?

File: modules/test/playwright/pages/portal-workflow-kaleo-designer-web/ActionPage.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	}
+
+	async fillWorkflowAction(name: string, script: string, typeOption: string) {
+		await this.nameInput.fill(name);
+		await this.selectActionType.selectOption(typeOption);
+		await this.scriptInput.fill(script);
+	}
+
+	async getTypeSelectOption(optionValue: string) {
+		return await this.page.$(`#type option[value="${optionValue}"]`);
```
Comment (cgmonte) [inline modules/test/playwright/pages/portal-workflow-kaleo-designer-web/ActionPage.ts:28]: The use of element handles is [discouraged by Playwright](https://playwright.dev/docs/api/class-elementhandle). Have you tried using Locators?

File: modules/test/playwright/pages/portal-workflow-kaleo-designer-web/NotificationPage.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+			await this.inputScript.fill(
+				(recipientTypeData as ScriptedRecipient)?.script
+			);
+		}
+	}
+
+	async getRecipientTypeTypeOption(optionValue: string) {
+		return await this.page.$(
+			`#recipient-type option[value="${optionValue}"]`
```
Comment (cgmonte) [inline modules/test/playwright/pages/portal-workflow-kaleo-designer-web/NotificationPage.ts:106]: The use of element handles is [discouraged by Playwright](https://playwright.dev/docs/api/class-elementhandle). Have you tried using Locators?

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hey, @danvb112, great work as always. I made some requests for change, please let me know if you have any questions or thoughts on this. I will be 100% available, as it is our priority to move this forward.

File: modules/test/playwright/tests/portal-security-script-management-web/scriptManagement.spec.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+		await expect(
+			scriptManagementPage.getActiveGroovyItemLocator(
+				'Groovy Object Action'
+			)
+		).toBeVisible();
+
+		await scriptManagementPage.doneButton.click();
+
+		await expect(scriptManagementPage.allowScriptCheckbox).toBeChecked();
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/test/playwright/tests/portal-security-script-management-web/scriptManagement.spec.ts:75]: Hey @cgmonte, It's really a funny thing! But in this case the product's decision was to just check if there are scripts in the application using the Save button, the Done button is only used to close the modal and reset the checkbox state to checked, always. This modal will only work if the checkbox is false and has scripts on the portal, so we only have to wait at the end to validate that after clicking on the Done button the checkbox will return to the marked state. Do you think I need to change anything in this test?

File: modules/test/playwright/tests/portal-security-script-management-web/scriptManagement.spec.ts [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+			objectDefinition.externalReferenceCode,
+			{
+				active: true,
+				engine: 'groovy',
+				engineLabel: 'Groovy',
+				errorLabel: {
+					en_US: 'Groovy Object Validation Error',
+				},
+				name: {
+					en_US: 'Groovy Object Validation',
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/test/playwright/tests/portal-security-script-management-web/scriptManagement.spec.ts:103]: @cgmonte Thanks for noticing this, I changed the object action and validation names to be random

File: modules/test/playwright/tests/portal-security-script-management-web/scriptManagement.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (cgmonte) [reply modules/test/playwright/tests/portal-security-script-management-web/scriptManagement.spec.ts:75]: Yeah, there is no reason to change this in the context of the test.

**Comment by danvb112** [comment]
Hey @cgmonte, thank you for this review, I did all the things you've suggested :smile:

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:playwright

**Comment by danvb112** [comment]
ci:test:sf

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by cgmonte** [comment]
ci:test:playwright

**Comment by danvb112** [comment]
ci:test:stable

**Comment by danvb112** [comment]
ci:forward:force

**Pull request closed without merging**